docs/design.md

```diff
@@ -312,22 +319,21 @@ There are several scenarios in which transactions interact:
   transaction’s commit timestamp to indicate that when/if the
   transaction does commit, it should use a timestamp *at least* as
   high. However, if the write conflict is from an SSI transaction,
-  the reader must compare priorities. If it has the higher priority,
-  it pushes the transaction’s commit timestamp, as with SI (that
+  the reader must compare priorities. If the reader has the higher priority,
+  it pushes the transaction’s commit timestamp (that
   transaction will then notice its timestamp has been pushed, and
-  restart). If it has the lower priority, it retries itself using as
+  restart). If it has the lower or same priority, it retries itself using as
   a new priority `max(new random priority, conflicting txn’s
-  priority - 1)`.
-
-- **Writer encounters uncommitted write intent with lower priority**:
-  writer aborts the conflicting transaction.
+  priority - 1)`. why max and not min?
 
-- **Writer encounters uncommitted write intent with higher priority**:
-  the transaction retries, using as a new priority *max(new random
-  priority, conflicting txn’s priority - 1)*; the retry occurs after
-  a short, randomized backoff interval.
+- **Writer encounters uncommitted write intent**:
+  If the other write intent has been written by a transaction with a lower
+  priority, the writer aborts the conflicting transaction. If the write 
+  intent has a higher or equal priority the transaction retries, using as a new 
+  priority *max(new random priority, conflicting txn’s priority - 1)*;
+  the retry occurs after a short, randomized backoff interval.
 
-- **Writer encounters committed write intent or newer committed value**:
+- **Writer encounters newer committed write intent or committed value**:
```

Comment:
"Committed write intent" is a little confusing; it's a write intent that was written by a transaction which has since committed. As soon as this situation is discovered, the intent is resolved and becomes an ordinary committed value. This must be done regardless of the intent's timestamp, so "newer" was in the right place before.

It might be better to restructure this a bit to call out "resolve any outstanding write intents" as a distinct step, instead of inventing distinct categories for "committed write intents" and "uncommitted write intents".